src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs

```diff
@@ -383,11 +383,32 @@ private async Task WriteDropManifest(List<DownloadedBuild> downloadedBuilds, str
             }
 
             Directory.CreateDirectory(outputDirectory);
-            string outputPath = Path.Combine(outputDirectory, "manifest.txt");
+            switch(_options.OutputFormat)
+            {
+                case DarcOutputType.yaml:
+                    await WriteYamlDropManifest(downloadedBuilds, Path.Combine(outputDirectory, "manifest.yaml"));
+                    break;
+                case DarcOutputType.json:
+                    await WriteJsonDropManifest(downloadedBuilds, Path.Combine(outputDirectory, "manifest.json"));
+                    break;
+                default:
+                    throw new NotImplementedException($"Darc output type {_options.OutputFormat} not supported in manifest generation");
+            }
+        }
+
+        /// <summary>
+        ///     Write the manifest in yaml format
+        /// </summary>
+        /// <param name="downloadedBuilds">List of downloaded builds in the manifest</param>
+        /// <param name="outputPath">Output file path of the manifest</param>
+        /// <returns>Async task</returns>
+        private async Task WriteYamlDropManifest(List<DownloadedBuild> downloadedBuilds, string outputPath)
```

Comment:
[nit] async methods should have `Async` suffix to help in comprehension of the API